server/poll/poll.go

```diff
@@ -108,26 +148,69 @@ func (p *Poll) AddAnswerOption(newAnswerOption string) *ErrorMessage {
 }
 
 // UpdateVote performs a vote for a given user
-func (p *Poll) UpdateVote(userID string, index int) error {
+func (p *Poll) UpdateVote(userID string, index int) (*i18n.Message, error) {
 	if len(p.AnswerOptions) <= index || index < 0 {
-		return fmt.Errorf("invalid index")
+		return nil, fmt.Errorf("invalid index")
+	}
+	if userID == "" {
+		return nil, fmt.Errorf("invalid userID")
+	}
+
+	if p.Settings.MaxVotes > 1 {
+		// Multi Answer Mode
+		votedAnswers, err := p.GetVotedAnswers(userID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get existing data")
+		}
+		for _, answers := range votedAnswers {
+			if answers == p.AnswerOptions[index].Answer {
+				return &i18n.Message{
+					ID:    "poll.updateVote.alreadyVoted",
+					Other: "You've already voted for this option.",
+				}, nil
+			}
+		}
+
+		if p.Settings.MaxVotes <= len(votedAnswers) {
+			return &i18n.Message{
+				ID:    "poll.updateVote.maxVotes",
+				Other: "You could't vote for this option, because you don't have any votes left. Use the reset button to reset your votes.",
+			}, nil
+		}
+	} else {
+		// Single Answer Mode
+		for _, o := range p.AnswerOptions {
+			for i := 0; i < len(o.Voter); i++ {
+				if userID == o.Voter[i] {
+					o.Voter = append(o.Voter[:i], o.Voter[i+1:]...)
+				}
+			}
+		}
 	}
+
+	p.AnswerOptions[index].Voter = append(p.AnswerOptions[index].Voter, userID)
+	return nil, nil
+}
+
+// ResetVotes remove votes by a given user
+func (p *Poll) ResetVotes(userID string) error {
```

Comment:
Similar comment as for `UpdateVote`: Is `error` really valuable here?

Reply:
It doesn't look valuable, so remove it.

But as for `UpdateVote`, it should check if  optionNumber is valid, because there may be the potential for inconsistency in the matterpoll and the potential for terrible errors if the value is incorrect. Also, if don't check the user ID, the user ID as empty will be added as a voter. (I was able to notice this by tests)